sleap_roots/pipelining.py

```diff
@@ -0,0 +1,625 @@
+            "base_ct": base_ct,
+            "base_top": base_top,
+            "base_deep": base_deep,
+            "base_length": base_length,
+            "base_length_ratio": base_length_ratio,
+            "base_meanx": base_meanx,
+            "base_meany": base_meany,
+            "base_medianx": base_medianx,
+            "base_mediany": base_mediany,
+            "base_stdx": base_stdx,
+            "base_stdy": base_stdy,
+            "pr_tip_depth": pr_tip_depth,
+            "base_median_ratio": base_median_ratio,
+            "base_ct_density": base_ct_density,
+            "tip_ct": tip_ct,
+            "tip_top": tip_top,
+            "tip_deep": tip_deep,
+            "tip_meanx": tip_meanx,
+            "tip_meany": tip_meany,
+            "tip_medianx": tip_medianx,
+            "tip_mediany": tip_mediany,
+            "tip_stdx": tip_stdx,
+            "tip_stdy": tip_stdy,
+            "tip_sdx_sdys": tip_sdx_sdys,
+            "tip_median_ratio": tip_median_ratio,
+            "conv_perimeters": conv_perimeters,
+            "conv_areas": conv_areas,
+            "conv_longest_dists": conv_longest_dists,
+            "conv_shortest_dists": conv_shortest_dists,
+            "conv_median_dists": conv_median_dists,
+            "conv_max_widths": conv_max_widths,
+            "conv_max_heights": conv_max_heights,
+            "ellipse_a": ellipse_a,
+            "ellipse_b": ellipse_b,
+            "ellipse_ratio": ellipse_ratio,
+            "network_width_depth_ratio": network_width_depth_ratio,
+            "network_solidity": network_solidity,
+            "network_length_lower_network": network_length_lower_network,
+            "network_distribution_ratio": network_distribution_ratio,
+            "count_scanline_interaction_max": count_scanline_interaction_max,
+            "count_scanline_interaction_min": count_scanline_interaction_min,
+            "count_scanline_interaction_mean": count_scanline_interaction_mean,
+            "count_scanline_interaction_std": count_scanline_interaction_std,
+            "count_scanline_interaction_median": count_scanline_interaction_median,
+            "scanline_start": scanline_start,
+            "scanline_end": scanline_end,
+        }
+    )
+
+    # append base and tip percentile data
+    for pctl in range(len(pctl_base)):
+        df["base_pctl_" + str(pctl_base[pctl])] = base_pctl[pctl]
+    for pctl in range(len(pctl_tip)):
+        df["tip_pctl_" + str(pctl_tip[pctl])] = tip_pctl[pctl]
+
+    return df
+
+
+def get_traits_plant(
+    path,
+    h5,
+    rice=False,
+    tolerance=0.02,
+    pctl_base=[25, 75],
+    pctl_tip=[25, 75],
+    fraction: float = 2 / 3,
+    depth: int = 1080,
+    width: int = 2048,
+    n_line: int = 50,
+    write_csv=True,
+) -> Tuple:
+    """Extract plant-based traits based on SLEAP prediction *.h5 and *.slp files.
+
+    Args:
+        path: the work directory where store h5 and slp files.
+        h5: h5 file, SLEAP prediction.
```

Comment:
These two are redundant. Just infer the folder from the series path.

Also the `h5` file is the image series, not the predictions (those are `.slp`).

I'd prefer if we can have the `primary` and `lateral` strings as arguments to this function. We can have defaults, but we should retain flexibility here.